geomstats/special_orthogonal_group.py

```diff
@@ -480,165 +526,243 @@ def matrix_from_quaternion(self, quaternion):
-            mask_0 = gs.squeeze(mask_0, axis=1)
-            coef_1[mask_0] = (1 - angle[mask_0] ** 2 / 12
-                              - angle[mask_0] ** 4 / 720
-                              - angle[mask_0] ** 6 / 30240)
-            coef_2[mask_0] = (1 / 12 + angle[mask_0] ** 2 / 720
-                              + angle[mask_0] ** 4 / 30240
-                              + angle[mask_0] ** 6 / 1209600)
-
-            mask_pi = gs.isclose(angle, gs.pi)
-            mask_pi = gs.squeeze(mask_pi, axis=1)
-            delta_angle = angle[mask_pi] - gs.pi
-            coef_1[mask_pi] = (- gs.pi * delta_angle / 4
-                               - delta_angle ** 2 / 4
-                               - gs.pi * delta_angle ** 3 / 48
-                               - delta_angle ** 4 / 48
-                               - gs.pi * delta_angle ** 5 / 480
-                               - delta_angle ** 6 / 480)
-            coef_2[mask_pi] = (1 - coef_1[mask_pi]) / angle[mask_pi] ** 2
-
-            mask_else = ~mask_0 & ~mask_pi
-            coef_1[mask_else] = ((angle[mask_else] / 2)
-                                 / gs.tan(angle[mask_else] / 2))
-            coef_2[mask_else] = (1 - coef_1[mask_else]) / angle[mask_else] ** 2
-
-            jacobian = gs.zeros((n_points, self.dimension, self.dimension))
-
-            for i in range(n_points):
-                if left_or_right == 'left':
-                    jacobian[i] = (coef_1[i] * gs.identity(self.dimension)
-                                   + coef_2[i] * gs.outer(point[i], point[i])
-                                   + skew_matrix_from_vector(point[i]) / 2)
-
-                else:
-                    jacobian[i] = (coef_1[i] * gs.identity(self.dimension)
-                                   + coef_2[i] * gs.outer(point[i], point[i])
-                                   - skew_matrix_from_vector(point[i]) / 2)
-
-        else:
-            if left_or_right == 'right':
-                raise NotImplementedError(
-                    'The jacobian of the right translation'
-                    ' is not implemented.')
-            jacobian = self.matrix_from_rotation_vector(point)
-
-        assert jacobian.ndim == 3
+        if point_representation is None:
+            point_representation = self.point_representation
+        assert self.belongs(point, point_representation)
+
+        if point_representation == 'vector':
+            if self.n == 3:
+                point = self.regularize(
+                    point, point_representation=point_representation)
+                n_points, _ = point.shape
+
+                angle = gs.linalg.norm(point, axis=1)
+                angle = gs.expand_dims(angle, axis=1)
+
+                coef_1 = gs.zeros([n_points, 1])
+                coef_2 = gs.zeros([n_points, 1])
+
+                mask_0 = gs.isclose(angle, 0)
+                mask_0 = gs.squeeze(mask_0, axis=1)
+                coef_1[mask_0] = (1 - angle[mask_0] ** 2 / 12
```

Comment:
didn't we have these coefficients somewhere? We need to define these.

Reply:
We had them defined for the hypersphere and hyperbolic space for not here. My bad. It's corrected now.